lib/discordrb/data/message.rb

```diff
@@ -408,5 +375,32 @@ def to_message
     end
 
     alias_method :message, :to_message
+
+    # The user who sent this message. Will be a {Member} most of the time, a {User} in the case that the
+    # user has left the server, or a {Recipient} if the message was sent in a private channel.
+    # @return [Member, User, Recipient]
+    def author
+      author_id = @author_data['id'].to_i
+      return unless author_id
+
+      if @author_data['discriminator'] == ZERO_DISCRIM
+        # This is a webhook user! It would be pointless to try to resolve a member here, so we just create
+        # a User and return that instead.
+        LOGGER.debug("Webhook user: #{author_id}")
+        User.new(data['author'], @bot)
+      elsif @channel.private?
+        # Turn the message user into a recipient - we can't use the channel recipient
+        # directly because the bot may also send messages to the channel
+        Recipient.new(@bot.user(author_id), @channel, @bot)
+      else
+        author = @channel.server.member(author_id) || @bot.user(author_id)
```

Comment:
Similarly, we should use ensure_member/ensure_user

Reply:
`member` is cached by default but made the other changes